jsettlers.logic/src/main/java/jsettlers/ai/highlevel/AiStatistics.java

```diff
@@ -193,8 +236,33 @@ private void updateMapStatistics() {
 				if (landscapeGrid.getResourceAmountAt(x, y) > 0) {
 					EResourceType resourceType = landscapeGrid.getResourceTypeAt(x, y);
 					sortedResourceTypes[resourceType.ordinal].addNoCollission(x, y);
-					if (resourceType != EResourceType.FISH || landscapeGrid.getLandscapeTypeAt(x, y) == ELandscapeType.WATER1) {
+					if (resourceType != EResourceType.FISH) {
 						aiMapInformation.resourceAndGrassCount[mapInformationPlayerId][resourceType.ordinal]++;
+						if (player != null) {
+							playerStatistics[player.playerId].resourceCount[resourceType.ordinal]++;
+						} else {
+							resourceCountInDefaultPartition[resourceType.ordinal]++;
+						}
+					} else if (landscapeGrid.getLandscapeTypeAt(x, y) == ELandscapeType.WATER1) {
+						int fishMapInformationPlayerId = mapInformationPlayerId;
+						if (mapInformationPlayerId == aiMapInformation.resourceAndGrassCount.length - 1) {
+							fishMapInformationPlayerId = mapInformationPlayerIdOfPosition((short) (x + 3), y);
+							if (fishMapInformationPlayerId == aiMapInformation.resourceAndGrassCount.length - 1) {
+								fishMapInformationPlayerId = mapInformationPlayerIdOfPosition((short) (x - 3), y);
+								if (fishMapInformationPlayerId == aiMapInformation.resourceAndGrassCount.length - 1) {
+									fishMapInformationPlayerId = mapInformationPlayerIdOfPosition(x, (short) (y + 3));
+									if (fishMapInformationPlayerId == aiMapInformation.resourceAndGrassCount.length - 1) {
+										fishMapInformationPlayerId = mapInformationPlayerIdOfPosition(x, (short) (y - 3));
+									}
+								}
+							}
+						}
+						aiMapInformation.resourceAndGrassCount[fishMapInformationPlayerId][resourceType.ordinal]++;
+						if (fishMapInformationPlayerId != aiMapInformation.resourceAndGrassCount.length - 1) {
```

Comment:
```MapInformation.resourceAndGrassCount``` is an Array. Because there is no index -1 I use playerNumber + 1 as index for the default partition.

However I inverted the if statement to make it more readable.